lib/structured_data.rb

```diff
@@ -0,0 +1,73 @@
+require 'nokogiri'
+require 'erb'
+
+# This creates a Structured Data FAQ section for google search results
+# https://developers.google.com/search/docs/data-types/faqpage
+#
+# The page needs to be annotated with the attribute `structured_data`
+# and questions and answers formatted with a `.section-faq`.
+# See `/articles/faq-ssl-certificates/` for an example.
+class AddStructuredDataFilter < Nanoc::Filter
+
+  identifier :add_structured_data
+
+  def run(content, params={})
+    if @item[:structured_data]
+      doc = Nokogiri::HTML(content)
+      write_faqs(doc, collect_faqs(doc))
+      return doc.to_html
+    end
+    content
+  end
+
+  def write_faqs(doc, faqs)
+    head = doc.at('head')
+    template = ERB.new <<-eos
+    <script type="application/ld+json">
+    {
+      "@context": "https://schema.org",
+      "@type": "FAQPage",
+      "mainEntity": [
+        <% faqs.each_with_index do |faq, index| %>
+          {
+            "@type": "Question",
+            "name": "<%= faq[:question] %>",
+            "acceptedAnswer": {
+              "@type": "Answer",
+              "text": "<%= faq[:answer] %>"
+            }
+          }
+          <%= index != faqs.size - 1 ? "," : "" %>
```

Comment:
I think this is a little bit hard to read, but I don't have a proposal to make this better that I'm happy with...

Reply:
I am going to extract this in a method. That could help